flowetl/etl/etl/model.py

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Define a DB model for storing the process of ingestion
+"""
+
+import pendulum
+from pendulum.date import Date as pendulumDate
+
+from sqlalchemy import Column, String, DateTime, Date, Integer
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm.session import Session
+
+Base = declarative_base()
+
+# pylint: disable=too-few-public-methods
+class ETLRecord(Base):
+    """
+    DB Model for storing the process of ingestion
+    for found files.
+    """
+
+    __tablename__ = "etl_records"
+    __table_args__ = {"schema": "etl"}
+
+    id = Column(Integer, primary_key=True)
+    file_name = Column(String)
+    cdr_type = Column(String)
+    cdr_date = Column(Date)
+    state = Column(String)
+    timestamp = Column(DateTime)
+
+    def __init__(
+        self, *, file_name: str, cdr_type: str, cdr_date: pendulumDate, state: str
+    ):
+
+        # This should be set more globally - not using enums
+        # because don't have a migration strategy in place
+        allowed_states = ["ingest", "archive", "quarantine"]
+        allowed_cdr_types = ["calls", "sms", "mds", "topups"]
```

Comment:
If you use @greenape's ingenius trick of inheriting from `str` when defining the enum, you can still have all the convenience of accepting strings as arguments but nicer validation without having to explicitly check the types:
```python
from enum import Enum

class CDRType(str, Enum):
    CALLS = "calls"
    SMS = "sms"
    MDS = "mds"
    TOPUPS = "topups"
```
And then you can replace the validation below (`cdr_type not in allowed_cdr_types`) simply with `self.cdr_type = CDRType(cdr_type)`. The latter will raise an error if `cdr_type` is not one of the supported ones:
```python
>>> CDRType("foo")
ValueError: 'foo' is not a valid CDRType
```
And you also still get string comparison if you want to:
```python
>>> "sms" == CDRType("sms") == CDRType.SMS
True
```

Reply:
But why! That's just an overly complex way of doing exactly the same thing...

Reply:
It is a more visible and checkable way of doing the same thing, and moves the responsibility for checking the validity of the input out of the function, no?